CSharpBasics/CSharpOOP/Program.cs

```diff
@@ -0,0 +1,180 @@
+                        new Homework_1().createItemsPriceList(priceForEachPositions), discount, items);
+Console.WriteLine($"Total resal for all position with discount = {positionResult} EUR\n");
+
+#endregion
+
+#region Task_3
+decimal totalResultWithoutOptionalArg = new Homework_1().countTotalPriceWithDiscount(new Homework_1().createItemsList(nameOfPositions), new Homework_1().createItemsPriceList(priceForEachPositions), discount);
+Console.WriteLine($"Total resal for all position with discount = {totalResultWithoutOptionalArg} EUR");
+
+Console.WriteLine();
+
+decimal totalResultWithOptionalArg = new Homework_1().countTotalPriceWithDiscount(new Homework_1().createItemsList(nameOfPositions), new Homework_1().createItemsPriceList(priceForEachPositions), discount, items);
+Console.WriteLine($"Total resal for all position with discount = {totalResultWithOptionalArg} EUR");
+#endregion
+
+#region Task_4
+
+Console.WriteLine("Enter five numbers:");
+double sideOne = double.Parse(Console.ReadLine());
+double sideTwo = double.Parse(Console.ReadLine());
+double sideThree = double.Parse(Console.ReadLine());
+double sideFour = double.Parse(Console.ReadLine());
+double sideFive = double.Parse(Console.ReadLine());
+
+double trianglePerimeter = new Homework_1().perimeterCalculation(sideOne, sideTwo, sideThree);
+double rectanglePerimeter = new Homework_1().perimeterCalculation(sideOne, sideTwo, sideOne, sideTwo);
+double quadrilateralPerimeter = new Homework_1().perimeterCalculation(sideOne, sideTwo, sideThree, sideFive);
+double pentagonPerimeter = new Homework_1().perimeterCalculation(sideOne, sideTwo, sideThree, sideFour, sideFive);
+
+Console.WriteLine($"Triangle perimeter is {trianglePerimeter}. " +
+    $"\nRectangle perimeter is {rectanglePerimeter}. " +
+    $"\nQuadrilateral perimeter is {quadrilateralPerimeter}." +
+    $"\nPentagon perimeter is {pentagonPerimeter} ");
+
+#endregion
+
+#region Task_5
+
+double areaCirclResult = new Homework_1().areaCalculation(2.5);
+Console.WriteLine(areaCirclResult);
+
+double areaTreangelCalculation = new Homework_1().areaCalculation(new Homework_1().perimeterCalculation(sideOne, sideTwo, sideThree), sideOne, sideTwo, sideThree);
+Console.WriteLine(areaTreangelCalculation);
+#endregion
+
+
+
+public class Homework_1
+{
+    public int roundValue = 2;
+
+    public decimal countTotalPriceWithDiscount(decimal items, decimal itemPrice, decimal discount)
+    {
+        decimal totalPrice = Decimal.Round((items * itemPrice), roundValue, MidpointRounding.AwayFromZero);
+        decimal discountSum = Decimal.Round((totalPrice * (discount / 100)), roundValue, MidpointRounding.AwayFromZero);
+        decimal totalSumWithDiscount = totalPrice - discountSum;
+
+        return totalSumWithDiscount;
+    }
+
+    public decimal countTotalPriceWithDiscount(string[] itemName, decimal[] itemPrice, decimal discount, decimal items=5)
```

Comment:
This option is possible here only if we have same number of each item, and if user requires different number for each number?) You can simplify this method by passing only number of items, name, price, discount and then justt call this method 3 times :)